### ISSUE TITLE
Proper Microsoft.NET.Sdk detection

### ISSUE THREAD

**Issue by matthid** [fsprojects/Paket#2863]
### Description

Apparently we currently don't detect project files starting like:

```
<Project>
<Import Sdk="Microsoft.NET.Sdk" Project="Sdk.props" />
```

Those are valid project files, just like 

```
<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```

**Comment by JohanLarsson** [comment]
I discussed this with @sharwell and he replied:
> The project system looks for <TargetFrameworks?\b
> Regular expression
> It totally ignores that this is xml

**Comment by enricosada** [comment]
yes.
also 

```xml
<Import Project="$(something)\Microsoft.NET.Sdk.props" />
```

i dont remember the right, but is described in https://github.com/Microsoft/msbuild/issues/1400

**Comment by matthid** [comment]
> The project system looks for <TargetFrameworks?\b

oO

**Comment by enricosada** [comment]
> The project system looks for <TargetFrameworks?\b

this is wrong, because i can add that in an imported project (like a `.props` file) or conditional

**Comment by sharwell** [comment]
@enricosada I'm not sure which part you refer to. I double checked the internal implementation for how Visual Studio knows to open a project file in the new project system, and it's essentially the following on the content of the csproj file:

```
if (file.Contains("<TargetFramework>") || file.Contains("<TargetFrameworks>"))
{
  // use new project system
}
else
{
  // use old project system
}
```

**Comment by 0x53A** [comment]
@sharwell I can confirm that loading these two files into VS15.5.0 Preview2 will trigger the OLD Project System:

__Directory.Build.props__

```
<Project>

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>
  
</Project>
```

__dotnet-test.csproj__

```
<Project Sdk="Microsoft.NET.Sdk">
  <ItemGroup>
    <PackageReference Include="Newtonsoft.Json">
      <Version>10.0.3</Version>
    </PackageReference>
  </ItemGroup>
</Project>
```

![image](https://user-images.githubusercontent.com/4236651/32126997-251392e4-bb74-11e7-9ba0-dd609109b85e.png)

![image](https://user-images.githubusercontent.com/4236651/32127015-3749a0ac-bb74-11e7-9858-74b6fcb64ac1.png)

**Comment by enricosada** [comment]
@sharwell nice to know.
but like that miss some scenario, like `TargetFramework` is defined in an imported .props file, or by env var. Not really common, i must admit :D

Maybe we are just too strict with detection, but i think trying to find if Sdk is used, is better (and doesnt add a lot burden)

**Comment by forki** [comment]
Tbh I think that's pretty common in our company.

Am 30.10.2017 10:12 schrieb "Enrico Sada" <notifications@github.com>:
